lib/find-visualstudio.js

```diff
@@ -113,9 +114,25 @@ class VisualStudioFinder {
     throw new Error('Could not find any Visual Studio installation to use')
   }
 
+  // Invoke the PowerShell script to get information about Visual Studio 2019
+  // or newer installations
+  async findVisualStudio2019OrNewer () {
+    return this.findNewVS()
+  }
+
+  // Invoke the PowerShell script to get information about Visual Studio 2017
+  async findVisualStudio2017 () {
+    if (this.nodeSemver.major >= 22) {
+      this.addLog(
+        'not looking for VS2017 as it is only supported up to Node.js 21')
+      return null
+    }
+    return this.findNewVS()
+  }
+
   // Invoke the PowerShell script to get information about Visual Studio 2017
   // or newer installations
-  async findVisualStudio2017OrNewer () {
+  async findNewVS () {
```

Comment:
The body of this function isn't changed. Wouldn't it still find VS2017 if that version is installed?

Reply:
This is just the implementation used for finding both VS2017 and later (2019/2022). It is called a few lines above, but there is a guard to fail for VS2017 if Node.js is v22+. This is [the code](https://github.com/JaneaSystems/node-gyp/blob/f38e957b6ee890540cd089575cb215e1b90c2fb1/lib/find-visualstudio.js#L119-L131) to do that,

Reply:
Oh that, yes, you're right! My bad, I'll fix it.

Reply:
@targos I've updated the code to address the issue you've found. Once it is approved, I'll just squash it and force-push. For now, it is separated for easier review.